source/frontend/src/features/documents/documentUpload/DocumentUploadContainer.tsx

```diff
@@ -62,30 +62,45 @@ export const DocumentUploadContainer: React.FunctionComponent<IDocumentUploadCon
     Api_Storage_DocumentTypeMetadataType[]
   >([]);
 
+  const onDocumentTypeChange = async (changeEvent: ChangeEvent<HTMLInputElement>) => {
+    const documentTypeId = Number(changeEvent.target.value);
+    await updateDocumentType(documentTypes.find(x => x.id === documentTypeId));
+  };
+
+  const updateDocumentType = useCallback(
+    async (documentType?: Api_DocumentType) => {
+      if (documentType === undefined) {
+        return;
+      }
+
+      setDocumentType(documentType.id?.toString() || '');
+      if (documentType.mayanId) {
+        const results = await retrieveDocumentTypeMetadata(documentType.mayanId);
+        if (results?.status === ExternalResultStatus.Success) {
+          setDocumentTypeMetadataTypes(results?.payload?.results);
+        }
+      } else {
+        setDocumentTypeMetadataTypes([]);
+      }
+    },
+    [retrieveDocumentTypeMetadata],
+  );
+
   useEffect(() => {
     const fetch = async () => {
       const axiosResponse = await retrieveDocumentTypes();
       if (axiosResponse && isMounted()) {
-        setDocumentTypes(axiosResponse);
+        if (props.relationshipType === DocumentRelationshipType.TEMPLATES) {
+          setDocumentTypes(axiosResponse.filter(x => x.documentType === 'CDOGS Template'));
```

Comment:
nit: would be nice to use a constant for this.

Reply:
sure thing.